weblate/trans/forms.py

```diff
@@ -904,12 +904,21 @@ def __init__(self, obj, user=None, *args, **kwargs):
         self.obj = obj
 
         # Add components from other projects with enabled shared TM
-        self.components = obj.project.component_set.filter(
-            source_language=obj.source_language
-        ) | Component.objects.filter(
-            source_language_id=obj.source_language_id,
-            project__contribute_shared_tm=True,
-        ).exclude(project=obj.project)
+        if isinstance(obj, Component) and hasattr(obj, "source_language"):
+            self.components = obj.project.component_set.filter(
+                source_language=obj.source_language
+            ) | Component.objects.filter(
+                source_language_id=obj.source_language_id,
+                project__contribute_shared_tm=True,
+            ).exclude(project=obj.project)
+
+        if hasattr(obj, "project") and hasattr(obj.project, "source_language_ids"):
+            self.components = obj.project.component_set.filter(
+                source_language_id__in=obj.project.source_language_ids
+            ) | Component.objects.filter(
+                source_language_id__in=obj.project.source_language_ids,
+                project__contribute_shared_tm=True,
+            ).exclude(project=obj.project)
```

Comment:
This needs more design decisions how it should behave. Presently, `obj` will be either `Component` or `None` because that is passed here:

https://github.com/WeblateOrg/weblate/blob/31358338c3613433794d19308c86269514232eda/weblate/addons/forms.py#L459

So this needs to be changed to pass `addon.instance.component or addon.instance.project` instead.

This form is also used for directly invoked automatic translation, where `obj` is always `Component` and the existing behavior should be preserved.

I think in the end we need something like this:

```suggestion
        if isinstance(obj, Component):
            obj = obj.project

        if obj is not None:
            # This is for both component and project scope
            self.components = obj.component_set.filter(
                source_language=obj.source_language
            ) | Component.objects.filter(
                source_language_id=obj.source_language_id,
                project__contribute_shared_tm=True,
            ).exclude(project=obj)
        else:
            # This is for site-wide
            self.components = Component.objects.all()
```

Reply:
After discussing, what we want for site wide addons

> we only offer site-wide machine translation engines in that case.

How to check for sitewide memory translation engine in case of Site-wide addons?

1. MachineryConfiguration has `sitewide` attribute - `sitewide` is a boolean field
2. machinery_settings is a dictionary of machinery settings - `sitewide` is a key in machinery_settings and what is the schema of machinery_settings?